replica/distributed_loader.cc

```diff
@@ -211,7 +212,14 @@ future<> run_resharding_jobs(sharded<sstables::sstable_directory>& dir, std::vec
         auto info_vec = std::move(reshard_jobs[this_shard_id()].info_vec);
         auto& cm = db.local().get_compaction_manager();
         auto max_threshold = table.schema()->max_compaction_threshold();
-        co_await d.reshard(std::move(info_vec), cm, table, max_threshold, creator);
+        auto local_owned_ranges_ptr = [&owned_ranges_ptr] () -> compaction::owned_ranges_ptr {
+            if (!owned_ranges_ptr) {
+                return nullptr;
+            }
+            auto owned_ranges = *owned_ranges_ptr;
+            return compaction::make_owned_ranges_ptr(std::move(owned_ranges));
+        };
```

Comment:
isn't owned ranges list immutable (lw_shared_ptr<const dht::token_range_vector>)? meaning we copy on write. why do we have to make a local copy?

Reply:
Eventually the compaction descriptor (per bucket on each shard) wants to keep a owned_ranges_ptr so we make a shard-local copy for calling sstable_directory::reshard.
I think we can push this decision there, by changing the interface to sstable_directory::reshard and pass it a `const token_range_vector&` so it can make the decision about making per-shard copies if needed on its own.

Reply:
In next version (v3), it is explicitly copied for creating a shard-local owned_ranges_ptr
and there's a comment explaining it.